src/main/scala/com/horizon/exchangeapi/AgbotsRoutes.scala

```diff
@@ -1291,13 +1291,13 @@ trait AgbotsRoutes extends JacksonSupport with AuthenticationSupport {
     exchAuth(TAgbot(compositeId),Access.WRITE) { _ =>
       validateWithMsg(reqBody.getAnyProblem) {
         complete({
-          db.run(AgbotAgreementsTQ.getNumOwned(compositeId).result.flatMap({ xs =>
+          val maxAgreements = ExchConfig.getInt("api.limits.maxAgreements")
+          val getNumOwnedDbio = if (maxAgreements == 0) DBIO.successful(0) else AgbotAgreementsTQ.getNumOwned(compositeId).result // avoid DB read for this if there is no max
+          db.run(getNumOwnedDbio.flatMap({ xs =>
             logger.debug("PUT /orgs/"+orgid+"/agbots/"+id+"/agreements/"+agrId+" num owned: "+xs)
```

Comment:
Wondering if this debug message will be confusing to someone looking at logs for the case where maxAgreements and maxMessages will be 0. The log message will indicate that "num owned" will be 0 when in fact the code didn't check the value. Same comment for the places in NodeRoutes

Reply:
Ah, good catch, i missed that one. I'll fix...